jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -94,22 +96,40 @@
     try (Timer ignored = new Timer(buildConfiguration.getBuildLogger(), DESCRIPTION)) {
       // Constructs the image.
       Image.Builder<CachedLayer> imageBuilder = Image.builder();
-      for (PullAndCacheBaseImageLayerStep pullAndCacheBaseImageLayerStep :
-          NonBlockingSteps.get(pullAndCacheBaseImageLayersStep)) {
+      Image<Layer> baseImage = NonBlockingSteps.get(pullBaseImageStep).getBaseImage();
+      ContainerConfiguration containerConfiguration =
+          buildConfiguration.getContainerConfiguration();
+      String buildTime =
+          containerConfiguration == null
+              ? Instant.EPOCH.toString()
+              : containerConfiguration.getCreationTime().toString();
+
+      // Base image layers/configuration passthrough
+      List<PullAndCacheBaseImageLayerStep> baseImageLayers =
+          NonBlockingSteps.get(pullAndCacheBaseImageLayersStep);
+      for (PullAndCacheBaseImageLayerStep pullAndCacheBaseImageLayerStep : baseImageLayers) {
         imageBuilder.addLayer(NonBlockingSteps.get(pullAndCacheBaseImageLayerStep));
       }
-      for (BuildAndCacheApplicationLayerStep buildAndCacheApplicationLayerStep :
-          buildAndCacheApplicationLayerSteps) {
-        imageBuilder.addLayer(NonBlockingSteps.get(buildAndCacheApplicationLayerStep));
+      for (HistoryObjectTemplate historyObject : baseImage.getHistory()) {
+        imageBuilder.addHistory(historyObject);
       }
-
-      // Parameters that we passthrough from the base image
-      Image<Layer> baseImage = NonBlockingSteps.get(pullBaseImageStep).getBaseImage();
       imageBuilder.addEnvironment(baseImage.getEnvironment());
       imageBuilder.addLabels(baseImage.getLabels());
 
-      ContainerConfiguration containerConfiguration =
-          buildConfiguration.getContainerConfiguration();
+      // Add history elements if base layers/history sizes don't match
+      if (baseImage.getHistory().size() < baseImageLayers.size()) {
+        int sizeDifference = baseImageLayers.size() - baseImage.getHistory().size();
+        for (int count = 0; count < sizeDifference; count++) {
```

Comment:
History with `empty_layer=true` should not be taken into account when calculating the history objects to add (they don't correspond to a layer). This should also get a test to make sure this is accounted for correctly.